modules/integration/tests-common/admin-clients/src/main/java/org/wso2/mb/integration/common/clients/AndesJMSConsumer.java

```diff
@@ -0,0 +1,576 @@
+                // Use an asynchronous message listener
+                receiver.setMessageListener(this);
+            } else {
+                // Uses a thread to listen to messages
+                Thread consumerThread = new Thread(this);
+                consumerThread.start();
+            }
+        } else {
+            throw new AndesClientException("The connection, session and message receiver is not assigned.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stopClient(){
+        /**
+         * Using a separate thread as stopping the consumer on "onMessage" thread is not allowed.
+         */
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                if (null != connection && null != session && null != receiver) {
+                    try {
+                        log.info("Closing Consumer");
+                        if (ExchangeType.TOPIC == consumerConfig.getExchangeType()) {
+                            if (null != receiver) {
+                                TopicSubscriber topicSubscriber = (TopicSubscriber) receiver;
+                                topicSubscriber.close();
+                            }
+
+                            if (null != session) {
+                                TopicSession topicSession = (TopicSession) session;
+                                topicSession.close();
+                            }
+
+                            if (null != connection) {
+                                TopicConnection topicConnection = (TopicConnection) connection;
+                                topicConnection.close();
+                            }
+                        } else if (ExchangeType.QUEUE == consumerConfig.getExchangeType()) {
+                            if (null != receiver) {
+                                QueueReceiver queueReceiver = (QueueReceiver) receiver;
+                                queueReceiver.close();
+                            }
+
+                            if (null != session) {
+                                QueueSession queueSession = (QueueSession) session;
+                                queueSession.close();
+                            }
+
+                            if (null != connection) {
+                                QueueConnection queueConnection = (QueueConnection) connection;
+                                queueConnection.stop();
+                                queueConnection.close();
+                            }
+                        }
+
+                        receiver = null;
+                        session = null;
+                        connection = null;
+
+                        log.info("Consumer Closed");
+
+                    } catch (JMSException e) {
+                        log.error("Error in stopping client.", e);
+                        throw new RuntimeException("Error in stopping client.", e);
```

Comment:
When writing test cases, we intend to write test cases that are meant to fail. So by throwing we can catch that specific exception denoting that the test case will pass. 
Reference - http://testng.org/javadocs/org/testng/annotations/ExpectedExceptions.html